src/core/Global.ttslua

```diff
@@ -1035,3 +1039,92 @@ function titleSplash(scenarioName)
     soundCubeApi.playSoundByName("Deep Bell")
   end
 end
+
+---------------------------------------------------------
+-- Update notification related functionality
+---------------------------------------------------------
+
+-- grabs the latest mod version and release notes from GitHub (called onLoad())
+function getModVersion()
+  WebRequest.get(SOURCE_REPO .. '/modversion.json', compareVersion)
+end
+
+-- compares the modversion with GitHub and possibly shows the update notification
+function compareVersion(request)
+  if request.is_error then
+    log(request.error)
+    return
+  end
+
+  -- global variable to make it accessible for other functions
+  modMeta = JSON.decode(request.text)
+
+  -- stop here if on latest version or "don't show again" was clicked
+  if MOD_VERSION == modMeta["latestVersion"] then return end
+  
+  if dontNotify[modMeta["latestVersion"]] then return end
+
+  -- end here if no release notes are found
+  if modMeta["releaseNotes"][modMeta["latestVersion"]] == nil then
+    log("Release notes for latest version not found!")
+  else
+    updateNotificationLoading()
+
+    -- delay to avoid lagging during onLoad()
+    Wait.time(function() UI.show("FinnIcon") end, 1)
+  end
+end
+
+-- updates the XML update notification based on the mod metadata
+function updateNotificationLoading()
+  -- grab data
+  local highlights = modMeta["releaseNotes"][modMeta["latestVersion"]]["Highlights"]
+  local ui = UI.getXmlTable()
```

Comment:
Reading and writing the entire XML table is very inefficient.  You can use UI.setAttribute() to update only the specific values you care about.  This will also remove the need for find_tag_with_id, which it very inefficient itself -  it's an O(n) lookup, which makes the entire operation O(n^2)

Reply:
Wow that's an awesome suggestion! I didn't know these awesome functions before